pkg/workers/workers.go

```diff
@@ -193,17 +199,18 @@ func updateRecords(node *masa.OracleNode, workEvent db.WorkEvent) {
 		return
 	}
 
-	exists, err := db.ReadData(node, workEvent.CID)
-	if err != nil {
-		logrus.Errorf("Failed to read data for CID %s: %v", workEvent.CID, err)
-		return
-	}
+	exists, _ := db.ReadData(node, workEvent.CID)
+	// we don't need to check for err since !exists gives an err also - we only need to know if the record exists or not in this context
 	if exists == nil {
 		err := db.WriteData(node, workEvent.CID, workEvent.Payload)
 		if err != nil {
 			logrus.Errorf("Failed to write data for CID %s: %v", workEvent.CID, err)
 			return
 		}
+		err = node.PubSubManager.Publish(config.TopicWithVersion(config.BlockTopic), workEvent.Payload)
```

Comment:
@jdutchak can you confirm this bit? It looks good here, but just wondering how that bubbled up by rebasing